README.md

```diff
@@ -6,6 +6,17 @@
 [![Build Status](https://ci.appveyor.com/api/projects/status/github/invenia/JLSO.jl?svg=true)](https://ci.appveyor.com/project/invenia/JLSO-jl)
 [![Codecov](https://codecov.io/gh/invenia/JLSO.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/invenia/JLSO.jl)
 
-JLSO wraps other formats (currently [BSON](https://github.com/MikeInnes/BSON.jl)),
-while storing key metadata such as the version of julia and all involved packages
-that were used in creating the file.
+JLSO is a storage container for serialized Julia objects.
+At the top-level it is a BSON file,
+where it stores metadata about the system it was created on as well as a collection of objects (the actual data).
+
+Depending on configuration, those objects may themselves be stored as BSON sub-documents,
+or in the native Julia serialization format (default).
+It is fast and efficient to load just single objects out of a larger file that contains many objects.
+
+The metadata (always stored in BSON) includes the Julia version and the versions of all packagers installed.
+This means in the worst case you can installed everything again and replicate your system.
+(Extreme worst case senario, using a BSON reader from anyther programming language).
+
+Note: that if the amount of data you have to store is very small, relative to the metadata about your enviroment.
+Then is is a pretty suboptimal format.
```

Comment:
```suggestion
```

Reply:
Why?What is the harm of end of file newlines?
Atom inserts them automatically...

Reply:
i meant "applying this suggestion might leave two newlines at the end of the file instead of one"

Reply:
Oh right gotcha

Reply:
Pretty sure it does not